### PR TITLE
Create authenticationService, userService, userDao

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

File: src/main/java/mate/academy/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.get(User.class, id);
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a user by id: " + id, e);
+        }
+    }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (nacenik) [inline src/main/java/mate/academy/dao/impl/UserDaoImpl.java:49]: Don't forget about catch block

File: src/main/java/mate/academy/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM User u WHERE u.email = :email", User.class)
+                    .setParameter("email", email)
+                    .uniqueResultOptional();
+        } catch (SQLGrammarException e) {
```
Comment (nacenik) [inline src/main/java/mate/academy/dao/impl/UserDaoImpl.java:62]: sqlgrammar? 
```suggestion
        } catch (Exception e) {
```

File: src/main/java/mate/academy/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            return Optional.empty();
```
Comment (nacenik) [inline src/main/java/mate/academy/dao/impl/UserDaoImpl.java:63]: you don't need to return empty here. Just throw a custom exception.

**Review by nacenik — CHANGES_REQUESTED** [review]
Let's improve your solution

**Review by DaniilNepelev — APPROVED** [review]

**Review by vovasalyha — APPROVED** [review]

**Pull request closed without merging**